src/hostname/hostnamed.c

```diff
@@ -355,6 +361,54 @@ static int get_hardware_model(Context *c, char **ret) {
         return get_dmi_properties(c, STRV_MAKE_CONST("ID_MODEL_FROM_DATABASE", "ID_MODEL"), ret);
 }
 
+static int get_hardware_sku(Context *c, char **ret) {
+        _cleanup_free_ char *model = NULL, *sku = NULL;
+        int r;
+
+        r = get_dmi_property(c, "ID_SKU", &sku);
+        if (r < 0)
+                return r;
+
+        /* Suppress reporting the SKU field, if it's the same string as the
+         * model field, which it appears to be on various systems */
+        r = get_hardware_model(c, &model);
+        if (r < 0) {
+                if (r != -ENOENT)
+                        return r;
+        } else if (streq_ptr(sku, model))
+                return -ENOENT;
+
+        return strdup_to(ret, sku);
+}
+
+static int get_hardware_version(Context *c, char **ret) {
+        _cleanup_free_ *version = NULL;
+        int r;
+
+        r = get_dmi_property(c, "ID_HARDWARE_VERSION", &version);
+        if (r < 0)
+                return r;
+
+        /* Suppress reporting the version field, if it's the same string as the
+         * model or sku field, which it appears to be on various systems */
+        for (int i = 0; i < 2; i++) {
+                _cleanup_free_ char *value = NULL;
+
+                if (i == 0)
+                        r = get_hardware_model(c, &value);
+                else
+                        r = get_hardware_sku(c, &value);
+
```

Comment:
please drop this empty line, we try to keep our function calls and their immediate error handling together. i.e. it's great to put empty lines before func calls, or after handling func results, but please not between func calls and handling of the func results